src/core/data_types.rs

```diff
@@ -527,4 +545,46 @@ mod tests {
         assert_eq!(r.cycle_focus(Direction::Backward), Some(&1));
         assert_eq!(r.as_vec(), vec![1, 2, 3]);
     }
+
+    #[test]
+    fn element() {
```

Comment:
I also added some tests to the existing selector methods, because it wasn't much work.